tests/chains/ethereum/scripts/cmd/helper/chain.go

```diff
@@ -0,0 +1,212 @@
+		ICS20Bank:     *ics20bank,
+		PathEnd:       *pathEnd,
+	}
+}
+
+func (chain *Chain) TxOpts(ctx context.Context, index uint32) *bind.TransactOpts {
+	return makeGenTxOpts(big.NewInt(chain.chainID), chain.prvKey(index))(ctx)
+}
+
+func (chain *Chain) prvKey(index uint32) *ecdsa.PrivateKey {
+	key, ok := chain.keys[index]
+	if ok {
+		return key
+	}
+	key, err := wallet.GetPrvKeyFromMnemonicAndHDWPath(chain.mnemonicPhrase, fmt.Sprintf("m/44'/60'/0'/0/%v", index))
+	if err != nil {
+		panic(err)
+	}
+	chain.keys[index] = key
+	return key
+}
+
+func (chain *Chain) CallOpts(ctx context.Context, index uint32) *bind.CallOpts {
+	opts := chain.TxOpts(ctx, index)
+	return &bind.CallOpts{
+		From:    opts.From,
+		Context: opts.Context,
+	}
+}
+
+func (chain *Chain) LastHeader(ctx context.Context) (*gethtypes.Header, error) {
+	return chain.client.HeaderByNumber(ctx, nil)
+}
+
+func makeGenTxOpts(chainID *big.Int, prv *ecdsa.PrivateKey) func(ctx context.Context) *bind.TransactOpts {
+	signer := gethtypes.LatestSignerForChainID(chainID)
+	addr := gethcrypto.PubkeyToAddress(prv.PublicKey)
+	return func(ctx context.Context) *bind.TransactOpts {
+		return &bind.TransactOpts{
+			From:     addr,
+			GasLimit: 6382056,
+			Signer: func(address common.Address, tx *gethtypes.Transaction) (*gethtypes.Transaction, error) {
+				if address != addr {
+					return nil, errors.New("not authorized to sign this account")
+				}
+				signature, err := gethcrypto.Sign(signer.Hash(tx).Bytes(), prv)
+				if err != nil {
+					return nil, err
+				}
+				return tx.WithSignature(signer, signature)
+			},
+		}
+	}
+}
+
+func InitializeChains(configDir, simpleTokenAddress, ics20TransferBankAddress, ics20BankAddress string) (*Chain, *Chain, error) {
+	pathConfig, err := parsePathConfig(configDir)
+	if err != nil {
+		return nil, nil, err
+	}
+	chainConfigs, err := parseChainConfigs(configDir + "/chains")
+	if err != nil {
+		return nil, nil, err
+	}
+	src := chainConfigs[0]
+	dst := chainConfigs[1]
+	ethClientA, err := client.NewETHClient(src.Chain.RpcAddr)
+	if err != nil {
+		return nil, nil, err
+	}
+	ethClientB, err := client.NewETHClient(dst.Chain.RpcAddr)
+	if err != nil {
+		return nil, nil, err
+	}
+	chainA := NewChain(pathConfig.Src, *src, ethClientA, src.Chain.HdwMnemonic, simpleTokenAddress, ics20TransferBankAddress, ics20BankAddress)
+	chainB := NewChain(pathConfig.Dst, *dst, ethClientB, dst.Chain.HdwMnemonic, simpleTokenAddress, ics20TransferBankAddress, ics20BankAddress)
+
+	return chainA, chainB, nil
+}
+
+func parsePathConfig(configDir string) (*core.Path, error) {
+	files, err := os.ReadDir(configDir)
+	if err != nil {
+		return nil, err
+	}
+	var pathConfig core.Path
+	for _, f := range files {
+		pth := fmt.Sprintf("%s/%s", configDir, f.Name())
+		if f.IsDir() {
+			continue
+		}
+		byt, err := os.ReadFile(pth)
+		if err != nil {
+			return nil, err
+		}
+		if err := json.Unmarshal(byt, &pathConfig); err != nil {
+			return nil, err
+		}
+	}
```

Comment:
Looping here is meaningless.
I think `parsePathConfig` takes a file path instead of a directory path.

Reply:
Change to file path
https://github.com/datachainlab/yui-relayer-build/pull/5/commits/77d3bb83fa9a4f3cbc0439525b0d5fb390110aeb